### ISSUE TITLE
Facing problem in line no:22.....Cannot read properties of undefined(reading params)

### ISSUE THREAD

**Issue by Anik-hasan07** [meabhisingh/mernProjectEcommerce#13]
![Screenshot (4)](https://user-images.githubusercontent.com/77586742/173676488-46f465bf-b3d0-44a3-b789-d46d7de6b3ea.png)

**Comment by manish6022** [comment]
Did you pass the match as an object in productDetails.......like productDeatails({match}) ??

**Comment by Anik-hasan07** [comment]
![Screenshot (6)](https://user-images.githubusercontent.com/77586742/173681126-4b7b406b-1164-4ac0-a7ec-c670d96924de.png)

**Comment by Abhiraj-cuchd** [comment]
You have to use the property 'useParams', then you can directly initialize a constant using useParams and then call the id.
example:
import {useParams} from 'react';

const params = useParams();  //initialize the constant

useEffect(() => {
    dispatch(getProductDetails(params.id))
  }, [dispatch, params.id])      //directly use the constant to fetch the id of the product

**Comment by manish6022** [comment]
Yes, you can use const params = useParams() if you are using react-router-dom version 6. In v6 match.param doesn't work anymore. But if you are using below v6 then you can use match.param..

**Comment by Anik-hasan07** [comment]
Thank u @Abhiraj-cuchd sir  now its  working.
Also @manish6022 sir thanks.

**Comment by sayedulkrm** [comment]
@Abhiraj-cuchd  @Anik-hasan07 @manish6022 
Thank You for this topic and solution.

**Comment by BilawalArif** [comment]
> Thank u @Abhiraj-cuchd sir now its working. Also @manish6022 sir thanks.

can you share the solution with me?

**Comment by HetMamtora** [comment]
Error is with React Router Dom version.
You are using v6 and match is available in v5

either downgrade your dom to v5 using this
npm uninstall --legacy-peer-deps react-router-dom
npm i --legacy--peer-deps react-router-dom@5.2

and if you want to use v6 of dom, following is the updated code that I am sharing, you just have to just import useMatch from router dom and give url address to it. 

You are good to go.
"Further in future new fixes will be there keep mentioning them so that someone else does not get stuck at this one place."

![image](https://user-images.githubusercontent.com/104263376/230329088-4c8820cc-7386-4aba-8c40-5c2f577b6663.png)

**Comment by Anurag13975** [comment]
> Yes, you can use const params = useParams() if you are using react-router-dom version 6. In v6 match.param doesn't work anymore. But if you are using below v6 then you can use match.param..

but after doing that params error resolved but a new issue raised as below

Cannot read properties of undefined (reading 'images')
TypeError: Cannot read properties of undefined (reading 'images')

can you help plase?
![issue](https://github.com/meabhisingh/mernProjectEcommerce/assets/116947003/9c7bf08b-6ee7-41ed-a97c-abe06ddb9b41)
